### ISSUE TITLE
Wrong TS definitions

### ISSUE THREAD

**Issue by ksocha** [icd2k3/use-react-router-breadcrumbs#7]
Hi, thanks for the lib!

Currently, the hook is defined as:
```js
function useBreadcrumbs<P>(
  routes?: BreadcrumbsRoute[],
  options?: Options
): Array<React.ReactNode | string>;
```

I think it should be like:
```js
function useBreadcrumbs(
  routes?: BreadcrumbsRoute[],
  options?: Options
): Array<BreadcrumbsProps>;
```

**Comment by ksocha** [comment]
Actually, after investingating a little more:

`useBreadcrumbs` should return an array of what `render` function returns:
```
{
    match: { url: string },
    location: types.Location,
    key: string,
    breadcrumb: React.ReactNode
}
```

That being said. since source code is TS, why not just generate typings from the source during build? I would be happy to help.

**Comment by icd2k3** [comment]
Hey @ksocha, thanks for bringing this to my attention!

I'm still pretty new to TS, so any assistance and/or PRs would be much appreciated!

**Comment by ksocha** [comment]
@icd2k3 
For now I have only basic understanding of the library internals and I know it would be a major change, but are you familiar with [tsdx](https://github.com/jaredpalmer/tsdx)? Using it would simplify configuration a lot.

**Comment by ksocha** [comment]
I opened a PR #8 with basic config change to generate declaration file. Please feel free to make any modifications needed.

**Comment by icd2k3** [comment]
Thanks for the PR! Fixed as of version `1.0.4`
